components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1,1802 +1,1177 @@
-
-    ! If CLUBB is called, do not call vertical diffusion, but obukov length and
-    ! surface friction velocity still need to be computed.  In addition, 
-    ! surface fluxes need to be updated here for constituents 
-    if (do_clubb_sgs) then
-
-        call clubb_surface ( state, ptend, ztodt, cam_in, surfric, obklen)
-
-        ! Update surface flux constituents 
-        call physics_update(state, ptend, ztodt, tend)
+       call check_tracers_chng(state, tracerint, "chem_timestep_tend", nstep, ztodt, cam_in%cflx)
+    end if ! chem_is_active
 
-    else
-    if (l_vdiff) then
-
-       call t_startf('vertical_diffusion_tend')
-       call vertical_diffusion_tend (ztodt ,state ,cam_in%wsx, cam_in%wsy,   &
-            cam_in%shf     ,cam_in%cflx     ,surfric  ,obklen   ,ptend    ,ast    ,&
-            cam_in%ocnfrac  , cam_in%landfrac ,        &
-            sgh30    ,pbuf )
+  end if ! l_tracer_aero
 
-    !------------------------------------------
-    ! Call major diffusion for extended model
-    !------------------------------------------
-    if ( waccmx_is('ionosphere') .or. waccmx_is('neutral') ) then
-       call mspd_intr (ztodt    ,state    ,ptend)
-    endif
+  !-----------------------------------------------------------------------------
+  ! Vertical diffusion/pbl calculation - (pbl, free atmosphere and molecular)
+  !-----------------------------------------------------------------------------
 
-       call physics_update(state, ptend, ztodt, tend)
-       call t_stopf ('vertical_diffusion_tend')
-    
-    end if ! l_vdiff
-    endif
+  call vertical_diffusion_tend(ztodt, state, cam_in%wsx, cam_in%wsy, cam_in%shf, cam_in%cflx, &
+                               surfric, obklen, ptend, ast, cam_in%landfrac, sgh30, pbuf )
+  call physics_update(state, ptend, ztodt, tend)
+  
+  obklen(:) = 0.
+  surfric(:) = 0.
 
+  !-----------------------------------------------------------------------------
+  ! Rayleigh friction calculation
+  !-----------------------------------------------------------------------------
+  if (l_rayleigh) then
 
-if (l_rayleigh) then
-    !===================================================
-    ! Rayleigh friction calculation
-    !===================================================
-    call t_startf('rayleigh_friction')
     call rayleigh_friction_tend( ztodt, state, ptend)
     call physics_update(state, ptend, ztodt, tend)
-    call t_stopf('rayleigh_friction')
 
-    if (do_clubb_sgs) then
-      call check_energy_chng(state, tend, "vdiff", nstep, ztodt, zero, zero, zero, zero)
-    else
-#ifdef MAML
-      call check_energy_chng(state, tend, "vdiff", nstep, ztodt, cam_in%cflx(:,1), zero, &
-           zero, shfavg_in)
-#else
-      call check_energy_chng(state, tend, "vdiff", nstep, ztodt, cam_in%cflx(:,1), zero, &
-          zero, cam_in%shf)
-#endif
-    endif
-    
+    call check_energy_chng(state, tend, "vdiff", nstep, ztodt, zero, zero, zero, zero)    
```

Comment:
This call to `check_energy_chng` after Rayleigh friction has changed; before the final argument was sensible heat flux from `cam_in`, but now it's zero. Is this intentional? Why the change?

Reply:
ah, looks like I accidently kept the version that's used with clubb. I'll fix it.